ios/keyman/Keyman/Keyman/Classes/MainViewController/MainViewController.swift

```diff
@@ -371,11 +376,18 @@ class MainViewController: UIViewController, TextViewDelegate, UIActionSheetDeleg
     // Resize textView and infoView
     let mainScreen: CGRect = UIScreen.main.bounds
     let margin: CGFloat = 2.0
-    let barHeights: CGFloat = AppDelegate.statusBarHeight() + navBarHeight()
     let kbHeight: CGFloat = keyboardVisible ? textView.inputView?.frame.height ?? 0 : 0
     let width: CGFloat = mainScreen.size.width
-    let height: CGFloat = mainScreen.size.height - barHeights - kbHeight
 
+    // This sort of thing really should be handled by the iOS layout engine, not programatically
+    // as it has been.  The original code was written pre-iOS 11 & safe areas!
+    // But, in the interest of a quick pre-release fix...
+    let window = UIApplication.shared.windows.first
+    let topPadding = window?.safeAreaInsets.top ?? 0
+    let barHeights: CGFloat = (topPadding == 0 ? AppDelegate.statusBarHeight() : topPadding) + navBarHeight()
+    let height: CGFloat = mainScreen.height - barHeights - kbHeight
+
+    // Like really, this REALLY should be done with constraints.
```

Comment:
About this:  see #6666.

I actually fixed it during development of this PR before I arrived at the full solution... then figured out how to separate the changes out into the form seen here.